subsys/bluetooth/host/smp.c

```diff
@@ -5416,6 +5421,14 @@ int bt_conn_set_bondable(struct bt_conn *conn, bool enable)
 {
 	struct bt_smp *smp;
 
+	if (IS_ENABLED(CONFIG_BT_CLASSIC) && (conn->type == BT_CONN_TYPE_BR)) {
+		if (atomic_test_bit(conn->flags, BT_CONN_BR_BONDABLE) == enable) {
+			return -EALREADY;
+		}
+		atomic_set_bit_to(conn->flags, BT_CONN_BR_BONDABLE, enable);
+		return 0;
```

Comment:
This isn't thread safe

Suggest
```suggestion
		if (enable && atomic_test_and_set_bit(conn->flags, BT_CONN_BR_BONDABLE)) {
			return -EALREADY;
		} else if (!enable && atomic_test_and_clear_bit(conn->flags, BT_CONN_BR_BONDABLE)) {
			return -EALREADY;
		}
		
		return 0;
```

Reply:
Updated.